From 33338ace2e3b3cb0b77f4d879355bb19547b645e Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Mon, 24 Nov 2025 16:02:32 +0100
Subject: [PATCH] cloud-ssh-keys-fetcher: allow systemd-detect-virt to see
 container runtimes

systemd-detect-virt checks /run/systemd/container to sense the presence
of a container runtim. Since cloud-ssh-keys-fetcher depends on this
systemd helper to decide where to pull authorized keys from, add an
allow rule.

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 policy/modules/contrib/cloud-ssh-keys-fetcher.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/contrib/cloud-ssh-keys-fetcher.te b/policy/modules/contrib/cloud-ssh-keys-fetcher.te
index 211b7e7f9..ffc6aeda5 100644
--- a/policy/modules/contrib/cloud-ssh-keys-fetcher.te
+++ b/policy/modules/contrib/cloud-ssh-keys-fetcher.te
@@ -2,6 +2,7 @@ policy_module(cloud-ssh-keys-fetcher, 0.0.1)
 
 gen_require(`
 	type init_t;
+	type init_var_run_t;
 	type sshd_t;
 ')
 
@@ -34,6 +35,9 @@ fs_getattr_nsfs_files(cloud_ssh_keys_fetcher_t)
 # systemd-detect-virt: which cloud is this? check DMI product ID
 dev_read_sysfs(cloud_ssh_keys_fetcher_t)
 
+# systemd-detect-virt: which container runtime is this?
+search_dirs_pattern(cloud_ssh_keys_fetcher_t, init_var_run_t, init_var_run_t)
+
 # curl: find home directory
 auth_read_passwd(cloud_ssh_keys_fetcher_t)
 
-- 
2.51.2

